File: examples/governance_static_go/domain/retire_framework.go
package domain

import (
	"time"
	"fmt"
)

type RetireFramework struct {
	FrameworkId string `json:"framework_id"`
}

func (c RetireFramework) CommandName() string { return "RetireFramework" }

func (c RetireFramework) Execute(repo RegulatoryFrameworkRepository) (*RegulatoryFramework, *RetiredFramework, error) {
	existing, err := repo.Find(c.FrameworkId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("RegulatoryFramework not found: %s", c.FrameworkId)
	}
	if existing.Status != "active" {
		return nil, nil, fmt.Errorf("cannot RetireFramework: RegulatoryFramework is in %s state", existing.Status)
	}
	existing.Status = "retired"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := RetiredFramework{
		AggregateID: existing.ID,
		FrameworkId: c.FrameworkId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/commands/retire_framework.rb
module ComplianceDomain
  class RegulatoryFramework
    module Commands
      class RetireFramework
        include Hecks::Command
        emits "RetiredFramework"

        attr_reader :framework_id

        def initialize(framework_id: nil)
          @framework_id = framework_id
        end

        def call
          existing = repository.find(framework_id)
          if existing
            unless existing.status == "active"
              raise ComplianceDomain::Error, "Cannot RetireFramework: status must be 'active', got '#{existing.status}'"
            end
            RegulatoryFramework.new(
              id: existing.id,
              name: existing.name,
              jurisdiction: existing.jurisdiction,
              version: existing.version,
              effective_date: existing.effective_date,
              authority: existing.authority,
              requirements: existing.requirements,
              status: "retired"
            )
          else
            raise ComplianceDomain::Error, "RegulatoryFramework not found: #{framework_id}"
          end
        end
      end
    end
  end
end
